database/abusescanner.go

```diff
@@ -187,12 +216,29 @@ func (db *AbuseScannerDB) FindUnblocked() ([]AbuseEmail, error) {
 // FindUnfinalized returns the messages that have not been finalized.
 func (db *AbuseScannerDB) FindUnfinalized(mailbox string) ([]AbuseEmail, error) {
 	emails, err := db.findGeneric(bson.M{
-		"email_uid": bson.M{"$regex": primitive.Regex{
-			Pattern: fmt.Sprintf("^%v-", mailbox),
-		}},
-		"parsed":    true,
-		"blocked":   true,
-		"finalized": false,
+		"$and": []bson.M{
+			{
+				"email_uid": bson.M{"$regex": primitive.Regex{
+					Pattern: fmt.Sprintf("^%v-", mailbox),
+				}},
+				"parsed":  true,
+				"blocked": true,
+			},
+			// unfinalized emails are either:
+			// - emails without csam that have not been finalized
+			// - emails with csm that have been reported but not yet finalized
+			{"$or": []bson.M{
+				{
+					"parse_result.tags": bson.M{"$nin": []string{"csam"}},
+					"finalized":         false,
```

Comment:
I did it for readability because the comment above says
```
// unfinalized emails are either:
// - emails without csam that have not been finalized
// - emails with csm that have been reported but not yet finalized
```
so it matched 1-1, but updated it 👍